js/controller.js

```diff
@@ -830,12 +830,16 @@ OO.plugin('Html5Skin', function(OO, _, $, W) {
 
     onVcPlay: function(event, source) {
       this.state.currentVideoId = source;
-      if (this.state.adWasPaused && this.state.currentAdsInfo && this.state.currentAdsInfo.currentAdItem && this.state.currentAdsInfo.currentAdItem.ssai) {
-        this.state.adPauseDuration = Date.now() - this.state.adPausedTime;
-        //we calculate new ad end time, based on the time that the ad was paused.
-        this.state.adEndTime = this.state.adEndTime + this.state.adPauseDuration; //milliseconds
-        this.state.adWasPaused = false;
-        this.state.adPauseDuration = 0;
+      if (this.state.currentAdsInfo && this.state.currentAdsInfo.currentAdItem && this.state.currentAdsInfo.currentAdItem.ssai) {
+        //When pause come from ad source, for SSAI we still want source to be main, for consistency.
+        this.state.currentVideoId = OO.VIDEO.MAIN;
```

Comment:
@ipadilla4 this looks ok for a quick patch but I am concerned that it does not address the root cause. Can we find out why we are receiving video id for both `OO.VIDEO.MAIN` and `OO.VIDEO.AD` here? I suspect there is something we need to change in the event itself to distinguish between SSAI and CSAI ads.

Reply:
I found out that from core plugin we are sending this `OO.VIDEO.ADS` videoId in the event `willResumeAds`. I agree that we need to address the root cause so I will double check that we always get `OO.VIDEO.MAIN` when having an ssai ad.

Reply:
@ipadilla4 got it, I think the solution then is to have a check in willResumeAds if we are playing ssai ads, and not publish `OO.VIDEO.ADS` if that is the case. We might be able to use `notifySSAIAdPlaying` and `notifySSAIAdPlayed` to keep track of this, or `notifyLinearAdStarted` and `notifyLinearAdEnded` but we would have to modify those to differentiate ssai ads

Reply:
@dallen42 yes we are currently using `notifyLinearAdStarted` and `notifyLinearAdEnded`, but we don't send videoId as 'OO.VIDEO.ADS`, we only send ad properties when publishing those events. So I think for ssai ads we are good.